torch/_inductor/overrides.py

```diff
@@ -163,7 +164,7 @@ def rand_like(x, **kwargs):
     if isinstance(x, torch.fx.Proxy):
         # double check we don't FX trace this
         return x.tracer.create_proxy("call_function", rand_like, (x), kwargs)
-    assert kwargs.get("device", x.device) == x.device
+    assert decode_device(kwargs.get("device", x.device)) == x.device
```

Comment:
but device arg doesn't need to match x's device? I think if you rewrite `get_seed_offset` to take in device argument, you can lift this restriction.

Reply:
Oh, yes, let me update this.